src/Services/SwaggerService.php

```diff
@@ -365,25 +380,33 @@ protected function savePostRequestParameters($actionName, $rules, array $annotat
                 ];
             }
 
-            $this->saveDefinitions($actionName, $rules, $annotations);
+            $this->saveDefinitions($actionName, $rules, $attributes, $annotations);
         }
     }
 
-    protected function saveDefinitions($objectName, $rules, array $annotations)
+    protected function saveDefinitions($objectName, $rules, $attributes, array $annotations)
     {
         $data = [
             'type' => 'object',
             'properties' => []
         ];
+
         foreach ($rules as $parameter => $rule) {
             $rulesArray = (is_array($rule)) ? $rule : explode('|', $rule);
             $parameterType = $this->getParameterType($rulesArray);
             $this->saveParameterType($data, $parameter, $parameterType);
-            $this->saveParameterDescription($data, $parameter, $rulesArray, $annotations);
+
+            $uselessRules = $this->rulesOfTypes;
+            $uselessRules['required'] = 'required';
 
             if (in_array('required', $rulesArray)) {
                 $data['required'][] = $parameter;
             }
+
+            $intersect = array_intersect_key(array_flip($rulesArray), $uselessRules);
```

Comment:
can we use an `arrar_diff` or `array_diff_assoc` functions?

Reply:
I think we can use `array_diff_key`